selection_sort.go

```diff
@@ -0,0 +1,16 @@
+package sort
```

Comment:
Precondition Check: The selection sort function should ensure that the input array is not nil before proceeding. Adding a check at the beginning of the function like `if arr == nil { return nil }` would prevent potential runtime errors.